ui/src/app/segments/segmentsApi.ts

```diff
@@ -0,0 +1,150 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
```

Comment:
should we follow the same pattern (using redux query) for flags and namespaces as well? seems pretty nice. maybe we should open issues to do that?

Reply:
It's a tough question as I haven't used RTK before. It looks nice but I expect that the biggest challenge will be the cache invalidation. It would be great to have more opinions about it from other contributors.

Reply:
Well, I have a different feelings about RTK. When you follow basics idea and patterns it shines. If you step away it's messy... Error handling is not easy and it's hard to transform them into unified format. I expected to remove lines, but I don't think it looks cleaner...

Reply:
yeah.. i can understand that. what do you think about instead using a similar pattern that we used in https://github.com/flipt-io/flipt/pull/2301 ? using async thunks to call the API layer that we have?

would that be too much of a refactor of this refactor?

Reply:
I tried it a bit before and decided to use RTK. It has other messy parts with Thunk and Selectors which I dislike. Like the `dispatch` in `useEffect`... it looks a bit wrong.  It has it's own challenges as we see from bug report by Mark Erikson.

Reply:
Makes sense. If we knew about RTK when doing the namespaces/flags we probably would have done it all this was as well as it does seem more natural 

Will give it a thorough look in the morning. Thanks for taking this on @erka !!